File: 0229-majority-element-ii/0229-majority-element-ii.java
class Solution {
    public List<Integer> majorityElement(int[] nums) {
        List<Integer> ans = new ArrayList<>();
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int num: nums){
            map.put(num, map.getOrDefault(num, 0)+1);
        }
        for(Map.Entry<Integer, Integer> es: map.entrySet()){
            if(es.getValue() > (nums.length/3)) ans.add(es.getKey());
        }
        return ans;
    }
}

File: 0229-majority-element-ii/0229-majority-element-ii.cpp
class Solution {
public:
    vector<int> majorityElement(vector<int>& nums) {
        vector<int> ans;
        map<int, int> m;
        for(int it:nums){
            m[it]++;
        }
        // for(auto it:m){
        //     cout << it.first << " -> " << it.second << endl; 
        // }
        for(auto it:m){
            if(it.second > nums.size()/3) ans.push_back(it.first);
        }
        return ans;
    }
};